tests/Integration/WorkloadManagementSystem/Test_TaskQueueDB.py

```diff
@@ -466,40 +464,103 @@ def test_chainWithTags():
+
+  # Matching only tq_job5 when Tag = ""
+  result = tqDB.matchAndGetTaskQueue({'Setup': 'aSetup', 'CPUTime': 50000,
+                                      'Tag': ''},
+                                     numQueuesToGet=5)
+  assert result['OK'] is True
+  # this should match whatever
+  assert int(result['Value'][0][0]) == tq_job5
+  assert len(result['Value']) == 1
+
+  # Matching only tq_job5 when Tag = []
+  result = tqDB.matchAndGetTaskQueue({'Setup': 'aSetup', 'CPUTime': 50000,
+                                      'Tag': []},
+                                     numQueuesToGet=5)
+  assert result['OK'] is True
+  # this should match whatever
+  assert int(result['Value'][0][0]) == tq_job5
+  assert len(result['Value']) == 1
+
   # Matching MultiProcessor
 
   # Tag: 'MultiProcessor'
-  # By doing this, we are basically saying that this CE is accepting MultiProcessor payloads
+  # By doing this, we are basically saying that this CE is accepting ALSO MultiProcessor payloads
   result = tqDB.matchAndGetTaskQueue({'Setup': 'aSetup', 'CPUTime': 50000,
                                       'Tag': 'MultiProcessor'},
                                      numQueuesToGet=4)
   assert result['OK'] is True
-  # this matches the tq_job1, as it is the only one that requires only MultiProcessor,
+  # this matches the tq_job1, as it is the only one that requires ONLY MultiProcessor,
   # AND the tq_job5, for which we have inserted no tags
-  # FIXME:-- is it correct?
   assert int(result['Value'][0][0]) in [tq_job1, tq_job5]
   assert len(result['Value']) == 2
 
+  # Tags: ['MultiProcessor', 'GPU']
+  # By doing this, we are basically saying that this CE is accepting ALSO payloads that require MultiProcessor or GPU
+  result = tqDB.matchAndGetTaskQueue({'Setup': 'aSetup', 'CPUTime': 50000,
+                                      'Tag': ['MultiProcessor', 'GPU']},
+                                     numQueuesToGet=4)
+  assert result['OK'] is True
+  # this matches the tq_job1, as it requires ONLY MultiProcessor
+  # the tq_job4, as it is the only one that requires BOTH MultiProcessor and GPU,
+  # AND the tq_job5, for which we have inserted no tags
+  # FIXME:-- is it correct?
+  assert int(result['Value'][0][0]) in [tq_job1, tq_job4, tq_job5]
+  assert len(result['Value']) == 3
+
   # RequiredTag: 'MultiProcessor'
   # By doing this, we are basically saying that this CE is accepting ONLY MultiProcessor payloads
   result = tqDB.matchAndGetTaskQueue({'Setup': 'aSetup', 'CPUTime': 50000,
                                       'RequiredTag': 'MultiProcessor'},
                                      numQueuesToGet=4)
   assert result['OK'] is True
-  # this matches the tq_job1, tq_job3, tq_job4 as these are those that expose the MultiProcessor tag (among others)
+  # this matches the tq_job1 as it is the only one that expose the MultiProcessor tag ONLY
+  # plus the tq_job3 and tq_job4 as they require ALSO the MultiProcessor tag
```

Comment:
This is not correct. The resource should also present Tag = Multiprocessor. RequiredTag does not replace Tag

Reply:
Are you saying that "RequiredTag" alone should not exist? What about BannedTag?

Reply:
If Tag is not published, jobs can not request it. If you specify BannedTag only, it makes no sense: jobs can not require it anyway. Specifying the same value in the Tag and in the BannedTag does not make sense. So, BannedTag is not really applicable at all. If you do not want jobs to choose some Tag, simply do not put it into the resource description.

RequiredTag should not exist alone. We can make a trick that RequiredTag assumes the same Tag value is defined but this is bad. Tag and RequiredTag have different semantics.